opentelekomcloud/config.go

```diff
@@ -198,6 +211,90 @@ func (c *Config) LoadAndValidate() error {
 	return nil
 }
 
+func newhwClient(c *Config) error {
+	ao := golangsdk.AuthOptions{
+		DomainID:         c.DomainID,
+		DomainName:       c.DomainName,
+		IdentityEndpoint: c.IdentityEndpoint,
+		Password:         c.Password,
+		TenantID:         c.TenantID,
+		TenantName:       c.TenantName,
+		TokenID:          c.Token,
+		Username:         c.Username,
+		UserID:           c.UserID,
+	}
+
+	client, err := huaweisdk.NewClient(ao.IdentityEndpoint)
+	if err != nil {
+		return err
+	}
+
+	// Set UserAgent
+	client.UserAgent.Prepend(terraform.UserAgentString())
```

Comment:
Just FYI this method is moving to [`httpclient.UserAgentString()`](https://github.com/hashicorp/terraform/blob/master/httpclient/useragent.go) but its also worth mentioning that currently this only shares the vendored TF version, not the CLI version used to invoke the provider, although we are looking to address that.

Reply:
I don't think you have to change this unless your vendored TF has the new method.